optuna_dashboard/ts/components/CompareStudies.tsx

```diff
@@ -0,0 +1,341 @@
+    const ids = queried
+      .split(",")
+      .map((s) => parseInt(s))
+      .filter((n) => !isNaN(n))
+    return studies.filter((t) => ids.findIndex((n) => n === t.study_id) !== -1)
+  }, [studies, query])
+}
+
+const getStudyListLink = (ids: number[]): string => {
+  const base = URL_PREFIX + "/compare-studies"
+  if (ids.length > 0) {
+    return base + "?ids=" + ids.map((n) => n.toString()).join(",")
+  }
+  return base
+}
+
+const isEqualDirections = (
+  array1: StudyDirection[],
+  array2: StudyDirection[]
+): boolean => {
+  let i = array1.length
+  if (i !== array2.length) return false
+
+  while (i--) {
+    if (array1[i] !== array2[i]) return false
+  }
+  return true
+}
+
+export const CompareStudies: FC<{
+  toggleColorMode: () => void
+}> = ({ toggleColorMode }) => {
+  const { enqueueSnackbar } = useSnackbar()
+  const theme = useTheme()
+  const query = useQuery()
+  const history = useHistory()
+
+  const action = actionCreator()
+  const studies = useRecoilValue<StudySummary[]>(studySummariesState)
+  const queried = useQueriedStudies(studies, query)
+  const selected = useMemo(() => {
+    return queried.length > 0 ? queried : studies.length > 0 ? [studies[0]] : []
+  }, [studies, query])
+
+  const studyListWidth = 200
+  const title = "Compare Studies"
+
+  useEffect(() => {
+    action.updateStudySummaries()
+  }, [])
+
+  const toolbar = (
+    <>
+      <IconButton
+        component={Link}
+        to={URL_PREFIX + "/beta"}
+        sx={{ marginRight: theme.spacing(1) }}
+        color="inherit"
+        title="Return to the top page"
+      >
+        <HomeIcon />
+      </IconButton>
+      <ChevronRightIcon sx={{ marginRight: theme.spacing(1) }} />
+      <Typography
+        noWrap
+        component="div"
+        sx={{ fontWeight: theme.typography.fontWeightBold }}
+      >
+        {title}
+      </Typography>
+    </>
+  )
+
+  return (
+    <Box sx={{ display: "flex" }}>
+      <AppDrawer toggleColorMode={toggleColorMode} toolbar={toolbar}>
+        <Box sx={{ display: "flex", flexDirection: "row", width: "100%" }}>
+          <Box
+            sx={{
+              minWidth: studyListWidth,
+              overflow: "auto",
+              height: `calc(100vh - ${theme.spacing(8)})`,
+            }}
+          >
+            <List>
+              <ListSubheader sx={{ display: "flex", flexDirection: "row" }}>
+                <Typography sx={{ p: theme.spacing(1, 0) }}>
+                  {studies.length} Studies
+                </Typography>
+                <Box sx={{ flexGrow: 1 }} />
+              </ListSubheader>
+              <Divider />
+              {studies.map((study, i) => {
+                return (
+                  <ListItem key={study.study_id} disablePadding>
+                    <ListItemButton
+                      onClick={(e) => {
+                        if (e.shiftKey) {
+                          let next: number[]
+                          const selectedNumbers = selected.map(
```

Comment:
Not a strong opinion but `selectedIds` might be clearer given the logic.
```suggestion
                          const selectedIds = selected.map(
```